### PR TITLE
Google loader

### PR DESCRIPTION
**Pull request by AshKyd**
Use Google Loader so we can:
- Load Google Maps
- Use the clientLocation to center on the user's rough location (IP-based geolocation).

Added associated tests etc.

### CODE REVIEW

**Comment by tatey** [comment]
Thanks for the patch! I'll take look ASAP.

**Comment by AshKyd** [comment]
PS, I'm not sure if you're aware, but the Location variable obscures the default browser window.Location. It's not causing any problems and I don't imagine it will in future, but food for thought. :)

**Comment by tatey** [comment]
I don't actually like the name `Location` so if we could come up with a better one I'd be happy to change it.

File: app/javascripts/application.js [hunk 1]
```diff
@@ -6,4 +6,33 @@
...
+    // Then load maps.
+    google.load("maps", "3", {
+      other_params:'sensor=false',
+      callback: function(){
+        // And initialize our app once it's done.
+        angular.bootstrap(document.querySelector('body'), ['am']);
+      }
+    });
+  });
+})();
```
Comment (tatey) [inline app/javascripts/application.js]: Missing an empty line at the EOF.

File: app/javascripts/services/google_location.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  if(window.google && google.loader && google.loader.ClientLocation && google.loader.ClientLocation.latitude){
+    center = {
+      lat: google.loader.ClientLocation.latitude,
+      lng: google.loader.ClientLocation.longitude
+    };
+  }
+  return {
+    center: center
+  }
+}]);
```
Comment (tatey) [inline app/javascripts/services/google_location.js:15]: Same here.

File: vendor/javascripts/google.loader-mocks.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+	}
+
+	// We don't need to do anything, it's already here!
+	if(google.loader){
+		return;
+	}
+
+	google.load = function(moduleName, moduleVersion, optionalSettings){
+		// Just enough time to simulate some async.
+		var timeoutLength = 100;
```
Comment (tatey) [inline vendor/javascripts/google.loader-mocks.js]: Given this exists to mock out the real service there is probably no need have an actual timeout. I think 0 would be fine.

File: vendor/javascripts/google.loader-mocks.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
+(function(){
```
Comment (tatey) [inline vendor/javascripts/google.loader-mocks.js:1]: This file doesn't really belong in vendor/javascripts but at the same time I can't think of a good place to keep it.

File: app/index.html [hunk 5]
```diff
@@ -69,6 +68,8 @@ <h4 class="list-group-item-heading" ng-bind="site.name"></h4>
     </div>
   </div>
 
+  <script type="text/javascript" src="https://www.google.com/jsapi"></script>
+  <script type="text/javascript" src="{% asset_path application.js %}"></script>
```
Comment (tatey) [inline app/index.html]: Don't really see the point of moving the JavaScript to the bottom of the document. Yes, I would do this in any normal website but this application is pretty much JavaScript only and there is no real advantage to having the DOM render without the JavaScript.

**Comment by tatey** [comment]
I'm glad we have a fallback for older browsers and are getting your current location from Google Maps.

Do you mind fixing the whitespace and setTimeout and I'll merge it in.

**Comment by AshKyd** [comment]
Cool. I think that's it. Have a sneaky peek and let me know what you think.

**Pull request closed without merging**